### PR TITLE
Fix linting errors in OAS files.

### PR DESCRIPTION
**Pull request by msporny**
This PR fixes linting errors in the OAS files. OAS linting requires that every endpoint define security properties and at least one server endpoint. Reviewers should pay specific attention to this PR because it does add authorization mechanisms available on each endpoint, including noAuth, didAuth, oAuth2, and zCap (since the current set of implementers support at least one of those mechanisms, if not multiple). Setting this PR to draft so we can discuss.

The intent of this PR is NOT to establish the correct authz mechanisms for each endpoint, that will be done in a future PR. The goal here is to get "close enough" and then fine tune the authz mechanisms in a future PR. Do not take the authz mechanisms provided as anything more than a work in progress at this point in time.

### CODE REVIEW

File: components/SecuritySchemes.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    oAuth2:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: /oauth2/authorize
+          tokenUrl: /oauth2/token
+          scopes:
+            read: Grants read access
+            write: Grants write access
```
Comment (msporny) [inline components/SecuritySchemes.yml:38]:
```suggestion
          authorizationUrl: /TBD
          tokenUrl: /TBD
          scopes:
            TBD: There are currently no scopes defined.
```

File: components/SecuritySchemes.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    didAuth:
+      type: http
+      scheme: signature
+      bearerFormat: VerifiablePresentation
+
+    oAuth2:
+      type: oauth2
+      flows:
+        authorizationCode:
```
Comment (msporny) [inline components/SecuritySchemes.yml:33]: You don't have to use authorizationCode -- need to figure out how to say "this is up to implementers".

File: components/SecuritySchemes.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  license:
+    name: W3C Software and Document License
+    url: http://www.w3.org/Consortium/Legal/copyright-software.
+  contact:
+    name: GitHub Source Code
+    url: https://github.com/w3c-ccg/vc-api
+paths:
+components:
+  securitySchemes:
+    noAuth:
```
Comment (msporny) [inline components/SecuritySchemes.yml:15]:
```suggestion
    firewall:
```

Comment (ottonomy) [reply components/SecuritySchemes.yml:15]: As noted on call, we understand the scope of this PR is to fix lint errors (we never want to compromise about dev tools not being broken), but we will update in future PRs to correctly make sure we don't unintentionally rule out use cases for "no authorization (header) required", "I enforce access to this endpoint with a firewall" or "some other secret auth method".

**Review by dlongley — CHANGES_REQUESTED** [review]
We should rename `oauth2` to `oauth2-vcapi` so that we can only define the parts we know we can define at this time. There were some requirements when using `oauth2` that weren't really ready to set, so if we invent a name for now as a placeholder we can just do the parts we have consensus on / know what we're doing.

File: components/SecuritySchemes.yml (hunk 1 above)
Comment (msporny) [inline components/SecuritySchemes.yml:38]:
```suggestion
    oAuth2:
      type: oauth2-vcapi
```

Comment (msporny) [reply components/SecuritySchemes.yml]: This has been removed in commit f2c17c0907ee42e5a819dfb858432688500ee20b.

File: components/SecuritySchemes.yml (hunk 2 above)
Comment (msporny) [reply components/SecuritySchemes.yml]: This has been removed in commit f2c17c0907ee42e5a819dfb858432688500ee20b and is up to the instance to define.

File: components/SecuritySchemes.yml (hunk 3 above)
Comment (msporny) [reply components/SecuritySchemes.yml]: This has been fixed in f2c17c0907ee42e5a819dfb858432688500ee20b.

**Comment by msporny** [comment]
> We should rename `oauth2` to `oauth2-vcapi` so that we can only define the parts we know we can define at this time. 

I was able to update it to just `oAuth2` and make the scheme `bearer` without defining all the oauth-y details in commit f2c17c0907ee42e5a819dfb858432688500ee20b.

**Review by dlongley — APPROVED** [review]

**Comment by msporny** [comment]
Discussed on the 2023-03-21 call.

Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**